presto-main/src/main/java/com/facebook/presto/execution/SqlTask.java

```diff
@@ -97,7 +97,16 @@ public SqlTask(
         requireNonNull(onDone, "onDone is null");
         requireNonNull(maxBufferSize, "maxBufferSize is null");
 
-        outputBuffer = new LazyOutputBuffer(taskId, taskInstanceId, taskNotificationExecutor, maxBufferSize, new UpdateSystemMemory(queryContext, taskId));
+        // pass a memory context supplier to the output buffer
+        // instead of a LocalMemoryContext as at this point
+        // we don't have the task context to get a local system
+        // memory context.
```

Comment:
```
// Pass a memory context supplier instead of memory context to the output buffer
// because we haven't created the task context that holds the the memory context.